rust/libnewsboat/src/history.rs

```diff
@@ -0,0 +1,142 @@
+use std::fs::OpenOptions;
+use std::io::{self, BufRead, BufReader, BufWriter, Write};
+use std::path::Path;
+
+pub struct History {
+    idx: usize,
+    lines: Vec<String>,
+}
+
+impl History {
+    pub fn new() -> History {
+        History {
+            idx: 0,
+            lines: Vec::new(),
+        }
+    }
+    pub fn add_line(&mut self, line: String) {
+        if !line.is_empty() {
+            self.lines.insert(0, line)
+        }
+        self.idx = 0;
+    }
+    pub fn next(&mut self) -> String {
+        match self.idx {
+            0 => String::from(""),
```

Comment:
It's better to use `String::new()` to avoid copying. Ditto a dozen lines below.

(I know nitpicks like that should be automated with `cargo clippy`, but I don't want to introduce it to codebase just yet because that makes it harder to contribute Rust code, and a good chunk of our contributors are new to Rust. The reason I decided to nitpick anyway is because you're already into Rust, so you might use a reminder about a more appropriate `String` method.)

Reply:
Definitely! I think I made the same mistake in `utils::get_basename`. I can fix those up in this PR as well? Are you saying your `cargo clippy` picked up on this? I might need to update my version of clippy.

Reply:
> I think I made the same mistake in utils::get_basename. I can fix those up in this PR as well?

Sure. But if you find more things to fix in the process, please make it a separate PR.

> Are you saying your cargo clippy picked up on this?j

Ah, no, I just assumed a linter would pick up on something like this.

Reply:
Ah. I'll leave it for another PR.